app/src/main/java/com/rpfcoding/snoozeloo/feature_alarm/scheduler_receiver/AlarmReceiver.kt

```diff
@@ -0,0 +1,171 @@
+package com.rpfcoding.snoozeloo.feature_alarm.scheduler_receiver
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.media.RingtoneManager
+import android.net.Uri
+import android.os.Handler
+import android.os.Looper
+import androidx.core.app.NotificationCompat
+import com.rpfcoding.snoozeloo.R
+import com.rpfcoding.snoozeloo.core.domain.ringtone.ALARM_MAX_REMINDER_MILLIS
+import com.rpfcoding.snoozeloo.core.util.isOreoPlus
+import com.rpfcoding.snoozeloo.core.util.isScreenOn
+import com.rpfcoding.snoozeloo.feature_alarm.domain.Alarm
+import com.rpfcoding.snoozeloo.feature_alarm.domain.AlarmConstants
+import com.rpfcoding.snoozeloo.feature_alarm.domain.AlarmRepository
+import kotlinx.coroutines.runBlocking
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import com.rpfcoding.snoozeloo.core.domain.ringtone.RingtoneManager as MyRingtoneManager
+
+class AlarmReceiver: BroadcastReceiver() {
+
+    private val helper by lazy { AlarmReceiverHelper() }
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        helper.onReceive(context, intent)
+    }
+
+    private class AlarmReceiverHelper: KoinComponent {
```

Comment:
`AlarmReceiver` acts as a delegate to the `AlarmReceiverHelper` class.
`AlarmReceiverHelper` can be removed.

At first I thought author didn’t know how to inject with Koin into a class that extends `BroadcastReceiver`, but he uses `KoinComponent` for the 2nd class.